rust/rust_c/src/trng.rs

```diff
@@ -0,0 +1,28 @@
+use core::num::NonZeroU32;
+
+use getrandom::{register_custom_getrandom, Error};
+
+use crate::bindings::GenerateTRNGRandomness;
+
+/// Same as in `keystore.h`.
+const ENTROPY_MAX_LEN: usize = 32;
+
+fn keystone_getrandom(dest: &mut [u8]) -> Result<(), Error> {
+    for chunk in dest.chunks_mut(ENTROPY_MAX_LEN) {
+        // SAFETY: `chunk.len()` is at most `ENTROPY_MAX_LEN` which fits in `u8`.
+        let len = chunk.len() as u8;
+
+        let ret = unsafe { GenerateTRNGRandomness(chunk.as_mut_ptr(), len) };
+
+        // TODO: Determine how to correctly map error codes from the underlying hardware
+        // into the `getrandom` custom error code space `Error::CUSTOM_START..=u32::MAX`.
+        if ret != 0 {
+            let error = NonZeroU32::new(Error::CUSTOM_START.saturating_add_signed(ret)).unwrap();
+            return Err(Error::from(error));
+        }
```

Comment:
The error codes here come from the underlying calls to `SE_GetDS28S60Rng` and `SE_GetAtecc608bRng`:
https://github.com/KeystoneHQ/keystone3-firmware/blob/08a44673b5fc6275bb71608a50b409849e134a3c/src/managers/keystore.c#L51-L91

I don't have access to their documentation and don't know what error codes they can return. This part of the code needs to be changed to be valid for the possible error codes, _or_ we can ignore the specific error codes and just return a fixed error code inside Rust.